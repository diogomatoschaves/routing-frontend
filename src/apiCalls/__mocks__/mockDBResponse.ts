const mockDBRoute = { 
  "id": "1234qwer", 
  "eta": 123, 
  "provider": {
    "name" : "Das",
    "type" : "routing",
    "url": "routing.das.car",
  },
  "source": {
    "name" : "Das1",
  },

  "waypoints": {
    "origin": [ 1.5, 0.11 ],
    "destination": [ 34.5, -112.11 ],
    "middlePoints" : [ [-34.15, -112.12] ]
  },
  "geometry": {
    "type": "LineString", 
    "coordinates": [ 
      [ 34.5, 12.11 ],
      [ -34.5, -12.11 ],
      [ 34.5, 112.11 ],
      [ 34.5, -12.11 ] 
  ]},
  "ata": 345,
  "confidence": 3,
  "date": "How I can check this" 
}


export const mockValidDBResponse = {
  exists: true,
  routes: [mockDBRoute]
}

export const mockInvalidDBResponse = {
  exists: false,
  routes: []
}

