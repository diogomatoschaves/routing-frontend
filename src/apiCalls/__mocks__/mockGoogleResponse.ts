const mockGoogleResponse = {
  geocoded_waypoints: [],
  routes: [{
    legs: [{
      duration: 10,
      distance: 1000
    }],
    overview_polyline: "ujtbFbsagVLvOD|@PnDJx@JT^?"
  }],
  duration: 10,
  distance: 1000,
}

export default mockGoogleResponse
