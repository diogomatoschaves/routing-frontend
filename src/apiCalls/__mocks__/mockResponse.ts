export const mockCarResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 86.5,
      totalDuration: 15.1,
      legs: [
        {
          duration: 15.1,
          distance: 86.5,
          geometry: [
            { lon: 13.389869, lat: 52.510348 },
            { lon: 13.389899, lat: 52.51035 },
            { lon: 13.390113, lat: 52.510363 },
            { lon: 13.390178, lat: 52.510367 },
            { lon: 13.390305, lat: 52.510375 },
            { lon: 13.390589, lat: 52.510393 },
            { lon: 13.390867, lat: 52.510409 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 }
          ]
        }
      ]
    }
  ],
  locations: [
    {
      snapDistance: 0.45025736761464097,
      location: {
        lon: 13.389869,
        lat: 52.510348
      }
    },
    {
      snapDistance: 12.073291914193243,
      location: {
        lon: 13.39114,
        lat: 52.510425
      }
    }
  ]
}

export const mockCarTrafficResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 86.5,
      totalDuration: 12.6,
      legs: [
        {
          duration: 12.6,
          distance: 86.5,
          geometry: [
            { lon: 13.389869, lat: 52.510348 },
            { lon: 13.389899, lat: 52.51035 },
            { lon: 13.390113, lat: 52.510363 },
            { lon: 13.390178, lat: 52.510367 },
            { lon: 13.390305, lat: 52.510375 },
            { lon: 13.390589, lat: 52.510393 },
            { lon: 13.390867, lat: 52.510409 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 }
          ]
        }
      ]
    }
  ],
  locations: [
    {
      snapDistance: 0,
      location: {
        lon: 13.389869,
        lat: 52.510348
      }
    },
    {
      snapDistance: 0,
      location: {
        lon: 13.39114,
        lat: 52.510425
      }
    }
  ]
}

export const mockFootResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 709.8,
      totalDuration: 514.9,
      legs: [
        {
          duration: 514.9,
          distance: 709.8,
          geometry: [
            { lon: 13.387123, lat: 52.527155 },
            { lon: 13.387215, lat: 52.527166 },
            { lon: 13.387215, lat: 52.527166 },
            { lon: 13.387189, lat: 52.52703 },
            { lon: 13.387184, lat: 52.527008 },
            { lon: 13.387283, lat: 52.526386 },
            { lon: 13.387283, lat: 52.526386 },
            { lon: 13.387743, lat: 52.526289 },
            { lon: 13.387824, lat: 52.526272 },
            { lon: 13.388361, lat: 52.526146 },
            { lon: 13.388507, lat: 52.526111 },
            { lon: 13.388802, lat: 52.526039 },
            { lon: 13.388987, lat: 52.525933 },
            { lon: 13.389304, lat: 52.525859 },
            { lon: 13.389668, lat: 52.525767 },
            { lon: 13.389995, lat: 52.525692 },
            { lon: 13.389995, lat: 52.525692 },
            { lon: 13.390095, lat: 52.525772 },
            { lon: 13.390474, lat: 52.525863 },
            { lon: 13.393425, lat: 52.526381 },
            { lon: 13.393541, lat: 52.526401 },
            { lon: 13.393625, lat: 52.526417 },
            { lon: 13.393977, lat: 52.526485 },
            { lon: 13.395811, lat: 52.526839 },
            { lon: 13.395811, lat: 52.526839 },
            { lon: 13.395811, lat: 52.526839 }
          ]
        }
      ]
    }
  ],
  locations: [
    { snapDistance: 4.3196444235727816, location: { lon: 13.387123, lat: 52.527155 } },
    { snapDistance: 1.0574398354756074, location: { lon: 13.395811, lat: 52.526839 } }
  ]
}
