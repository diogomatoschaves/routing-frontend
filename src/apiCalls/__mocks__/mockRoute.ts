export const mockRoute = [
  { lat: 52.510348, lon: 13.389869 },
  { lat: 52.51035, lon: 13.389899 },
  { lat: 52.510363, lon: 13.390113 },
  { lat: 52.510367, lon: 13.390178 },
  { lat: 52.510375, lon: 13.390305 },
  { lat: 52.510393, lon: 13.390589 },
  { lat: 52.510409, lon: 13.390867 },
  { lat: 52.510425, lon: 13.39114 },
  { lat: 52.510425, lon: 13.39114 },
  { lat: 52.510425, lon: 13.39114 }
]

export const mockGoogleRoute = [
  { lat: 37.30107, lon: -121.91042 },
  { lat: 37.301, lon: -121.9131 },
  { lat: 37.30097, lon: -121.91341 },
  { lat: 37.30088, lon: -121.91429 },
  { lat: 37.30082, lon: -121.91458 },
  { lat: 37.30076, lon: -121.91469 },
  { lat: 37.3006, lon: -121.91469 }
]
