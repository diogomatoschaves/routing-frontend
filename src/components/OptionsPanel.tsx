import React, { Component } from 'react'
import styled, { css } from 'styled-components'
import ProfileToggler from './ProfileToggler'
import JsonRenderer from './JsonRenderer'
import { Button, Dropdown, TextArea, Form, Header, Icon } from 'semantic-ui-react'
import { Box } from '../styledComponents'
import {
  RouteResponse,
  Body,
  UpdateState,
  UpdatePoint,
  Location,
  MatchResponse
} from '../types'
import { MAIN_PETROL, PETROL_3 } from '../utils/colours'
import { capitalize } from '../utils/functions'
import { Schema } from '../utils/schemas/index'
import { Validator } from 'jsonschema'
import JSON5 from 'json5'


interface Props {
  handleHideClick: (e: any) => void
  response: RouteResponse | MatchResponse
  body: Body | undefined
  endpoint: string
  updateState: UpdateState
  updatePoint: UpdatePoint
  responseOption: string
  locations: Array<Location>
  defaultColor: string
  selectedService: number
  serviceOptions: any
}

interface State {
  bodyValue: string
  responseValue: string
  validator?: Validator
  responseColor: string
  bodyColor: string
  responseEdit: boolean
  bodyEdit: boolean
}

const StyledForm = styled(Form)`
  width: 100%;
  z-index: 0;

  &.ui.form textarea {
    width: 100%;
    font-size: 16px;
    border-radius: 7px;
    border: none;
    color: ${props => (props.color ? props.color : 'rgb(100, 100, 100)')};
    background-color: rgb(242, 242, 242);
    user-select: text;
    z-index: 1000;

    &:focus {
      background-color: rgb(248, 248, 248);
      color: ${props => (props.focuscolor ? props.focuscolor : 'black')};
      /* border-radius: 7px; */
    }

    &::-webkit-scrollbar {
      width: 8px;
    }

    &::-webkit-scrollbar-track {
      -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
      border-radius: 10px;
    }

    &::-webkit-scrollbar-thumb {
      border-radius: 10px;
      -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.5);
    }
  }
`

const StyledHeader = styled(Header)`
  &.ui.header {
    margin-top: 10px;
    align-self: flex-start;
    ${(props: any) =>
      props.overridecolor &&
      css`
        color: ${props.overridecolor};
      `}
    &:first-child {
      margin-top: 12px;
    }
  }
`

const StyledDropdown = styled(Dropdown)`
  &.ui.selection.dropdown {
    color: rgb(100, 100, 100);
    border-color: rgb(242, 242, 242);
    background-color: rgb(242, 242, 242);
  }
`

const StyledButton = styled(Button)`
  &.ui.button {
    padding: 7px;
    ${props =>
      props.marginleft &&
      css`
        margin-left: ${props.marginleft};
      `}
    &.icon {
      background-color: ${MAIN_PETROL};
      color: white;
      padding: 7px;
    }
    
  }
`

export default class OptionsPanel extends Component<Props, State> {
  static defaultProps = {
    defaultColor: 'rgb(100, 100, 100)'
  }

  state = {
    bodyValue: '{}',
    responseValue: '{}',
    selectedService: 0,
    validator: new Validator(),
    responseColor: 'rgb(100, 100, 100)',
    bodyColor: 'rgb(100, 100, 100)',
    responseEdit: false,
    bodyEdit: false
  }

  componentDidMount() {
    const { validator } = this.state
    const { response, body } = this.props

    Promise.resolve(
      Object.keys(Schema.Route).forEach((obj: any) => {
        validator.addSchema(Schema.Route[obj], `/${obj}`)
      })
    ).then(() => {
      Object.keys(Schema.Match).forEach((obj: any) => {
        validator.addSchema(Schema.Match[obj], `/${obj}`)
      })
    })

    this.setState({ 
      validator, 
      bodyValue: JSON5.stringify(body, null, 2),
      responseValue: JSON5.stringify(response, null, 2), 
    })
  }

  componentDidUpdate(prevProps: Props, prevState: State) {
    const { body, response, defaultColor } = this.props
    const { responseEdit, bodyEdit } = this.state

    if (prevProps.body !== body || (prevState.bodyEdit !== bodyEdit && !bodyEdit)) {
      this.setState({
        bodyValue: JSON5.stringify(body, null, 2),
        bodyColor: defaultColor
      })
    }

    if (
      prevProps.response !== response ||
      (prevState.responseEdit !== responseEdit && !responseEdit)
    ) {
      this.setState({
        responseValue: JSON5.stringify(response, null, 2),
        responseColor: defaultColor
      })
    }
  }

  handleInputChange = ({ id, value }: { id: string; value: any }): void => {
    this.setState(state => ({
      ...state,
      [id]: value
    }))
  }

  handleBlur = ({ id, value }: { id: string; value: any }): void => {
    const { validator, selectedService } = this.state
    const {
      updateState,
      updatePoint,
      responseOption,
      locations,
      defaultColor,
      serviceOptions
    } = this.props

    const service = serviceOptions[selectedService].key
    this.processJSON(
      value,
      validator,
      updateState,
      updatePoint,
      service,
      id.includes('response') ? 'response' : 'body',
      responseOption,
      locations,
      defaultColor
    )
  }

  processJSON = (
    value: any,
    validator: Validator,
    updateState: UpdateState,
    updatePoint: UpdatePoint,
    service: string,
    inputType: string,
    responseOption: string,
    locations: Array<Location>,
    defaultColor: string
  ) => {
    let parsedValue
    try {
      parsedValue = JSON5.parse(value)
    } catch (error) {
        console.log('Invalid JSON')
        this.setState(state => ({
          ...state,
          [`${inputType}Color`]: 'red'
        }))
        return null
    }
    const validation = validator.validate(
      parsedValue,
      Schema[service][capitalize(inputType)]
    )
    if (validation.valid) {
      if (inputType === 'response') {
        this.processValidResponse(
          updateState,
          updatePoint,
          responseOption,
          locations,
          parsedValue
        )
      } else {
        this.processValidBody(updatePoint, locations, parsedValue)
      }

      this.setState(state => ({
        ...state,
        [`${inputType}Color`]: defaultColor
      }))
    } else {
      this.setState(state => ({
        ...state,
        [`${inputType}Color`]: 'red'
      }))
    }
  }

  processValidResponse = (
    updateState: UpdateState,
    updatePoint: UpdatePoint,
    responseOption: string,
    locations: Array<Location>,
    parsedValue: any
  ) => {
    const points: any = {
      start: parsedValue.locations[0].location,
      end: parsedValue.locations.slice(-1)[0].location
    }

    if (
      [locations[0], locations.slice(-1)[0]].some(el => {
        return el.lat !== points[el.name].lat || el.lng !== points[el.name].lon
      })
    ) {
      new Promise(resolve => {
        resolve(updateState('recalculate', false))
      }).then((value) => {
        updatePoint([0, 1], [
          { lat: points.start.lat, lng: points.start.lon },
          { lat: points.end.lat, lng: points.end.lon }
        ])
      })

      updateState(
        responseOption === 'normal' ? 'response' : 'trafficResponse',
        parsedValue
      )
    }
  }

  processValidBody = (
    updatePoint: UpdatePoint,
    locations: Array<Location>,
    parsedValue: any
  ) => {
    const points: any = {
      start: parsedValue.locations[0],
      end: parsedValue.locations.slice(-1)[0]
    }

    if (
      [locations[0], locations.slice(-1)[0]].some(el => {
        return el.lat !== points[el.name].lat || el.lng !== points[el.name].lon
      })
    ) {
      updatePoint([0, 1], [
        { lat: points.start.lat, lng: points.start.lon },
        { lat: points.end.lat, lng: points.end.lon }
      ])
    }
  }

  updateState = (id: string) => {
    this.setState((state: any) => ({
      ...state,
      [id]: !state[id]
    }))
  }

  render() {
    const {
      endpoint,
      updateState,
      responseOption,
      response,
      body,
      selectedService,
      serviceOptions
    } = this.props

    const {
      bodyValue,
      responseValue,
      responseColor,
      responseEdit,
      bodyEdit,
      bodyColor
    } = this.state

    return (
      <Box>
        <Box
          padding="30px 10px 0 0"
          height={'70px'}
          direction="row"
          justify="space-around"
        >
          <Box width="35%" padding="0 0 0 3%">
            <StyledDropdown
              placeholder="Select Service"
              id={'selectedService'}
              fluid
              selection
              options={serviceOptions}
              value={selectedService}
              onChange={(e: any, { id, value }: any) => updateState(id, value)}
            />
          </Box>
          <Box width="35%">
            {serviceOptions[selectedService].key === 'Route' && (
              <ProfileToggler
                updateState={updateState}
                id={'responseOption'}
                responseOption={responseOption}
              />
            )}
          </Box>
        </Box>
        <Box width="80%" padding="10px 0 10px 0">
          <StyledHeader overridecolor={MAIN_PETROL}>Request Endpoint</StyledHeader>
          <StyledForm focuscolor={PETROL_3}>
            <TextArea id={'endpointValue'} rows={1} value={endpoint} readOnly disabled />
          </StyledForm>
        </Box>
        <Box width="80%" padding="10px 0 10px 0">
          <Box direction="row" justify="flex-start">
            <Box direction="row" justify="flex-start">
              <StyledHeader overridecolor={MAIN_PETROL}>Request Body</StyledHeader>
              <StyledButton
                id={'bodyEdit'}
                onClick={(e: any, { id }: { id: string }) => this.updateState(id)}
                marginleft={'10px'}
                icon={bodyEdit}
                className="body-textarea"
              >
                {bodyEdit ? <Icon name={'check'} /> : 'Edit'}
              </StyledButton>
            </Box>
          </Box>
          <JsonRenderer
            value={body}
            editableValue={bodyValue}
            id={'bodyValue'}
            editJson={bodyEdit}
            handleBlur={this.handleBlur}
            handleInputChange={this.handleInputChange}
            rows={10}
            color={bodyColor}
          />
        </Box>
        <Box width="80%" padding="10px 0 10px 0">
          <Box direction="row" justify="flex-start">
            <StyledHeader overridecolor={MAIN_PETROL}>Response</StyledHeader>
            <StyledButton
              id={'responseEdit'}
              onClick={(e: any, { id }: { id: string }) => this.updateState(id)}
              marginleft={'10px'}
              icon={responseEdit}
              className="response-textarea"
            >
              {responseEdit ? <Icon name={'check'} /> : 'Edit'}
            </StyledButton>
          </Box>
          <JsonRenderer
            value={response}
            editableValue={responseValue}
            id={'responseValue'}
            editJson={responseEdit}
            handleBlur={this.handleBlur}
            handleInputChange={this.handleInputChange}
            rows={15}
            color={responseColor}
          />
        </Box>
      </Box>
    )
  }
}
