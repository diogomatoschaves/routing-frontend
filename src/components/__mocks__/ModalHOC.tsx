import React from 'react'

export default function MockModalHOC (Component: any) {
  return (
    <div>
      <Component />
    </div>
  )
}