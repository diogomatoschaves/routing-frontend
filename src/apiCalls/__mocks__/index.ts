import { mockCarResponse, mockFootResponse, mockCarTrafficResponse } from './mockResponse'
import mockGoogleResponse from './mockGoogleResponse'
import { mockValidDBResponse, mockInvalidDBResponse } from './mockDBResponse'

export const routingApi = jest.fn().mockImplementation((profile, authorization, body) => {
  if (body.locations[0].lat === 100) {
    return Promise.resolve({
      ...mockCarResponse,
      code: 'LocationInvalid'
    })
  }

  return profile === 'car'
    ? Promise.resolve(mockCarResponse)
    : profile === 'car-traffic'
    ? Promise.resolve(mockCarTrafficResponse)
    : Promise.resolve(mockFootResponse)
})

export const googleDirections = jest
  .fn()
  .mockImplementation((google, profile, locations) => {
    if (locations[0].lat === 100) {
      return Promise.resolve({
        ...mockGoogleResponse,
        status: 'ZERO_RESULTS'
      })
    }
    return Promise.resolve(mockGoogleResponse)
  })

export const fetchRouteDB = jest.fn().mockImplementation(routeId => {
  return routeId === 'AAA'
    ? Promise.resolve(mockValidDBResponse)
    : Promise.resolve(mockInvalidDBResponse)
})
