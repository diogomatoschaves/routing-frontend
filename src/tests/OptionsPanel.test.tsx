import React from 'react'
import TestRenderer from 'react-test-renderer'
import OptionsPanel from '../components/OptionsPanel'
import { MemoryRouter } from 'react-router-dom'
import { mockCarResponse } from '../apiCalls/__mocks__/mockResponse'



const mockLocations = [{ 
  name: 'start', 
  marker: 'map marker alternate',
  placeholder: 'Origin',
  lat: null,
  lng: null
}, { 
  name: 'end', 
  marker: 'flag checkered',
  placeholder: 'Destination',
  lat: null,
  lng: null
}]

const mockEventStart = { lng: 13.389869, lat: 52.510348 }
const mockEventEnd = { lng: 13.39114, lat: 52.510425 }

const mockBody = {
  locations: [
    { 
      lat: mockEventStart.lat,
      lon: mockEventStart.lng
    },
    { 
      lat: mockEventEnd.lat,
      lon: mockEventEnd.lng
    },
  ],
  reportGeometry: true,
}

const serviceOptions = [
  {
    key: 'Route',
    text: 'Route',
    value: 0
  },
  {
    key: 'Match',
    text: 'Match',
    value: 1
  }
]

const endpoint = 'https://routing.develop.otonomousmobility.com'

it('expect to render OptionsPanel component', () => {

  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <OptionsPanel 
        handleHideClick={jest.fn()}
        response={mockCarResponse}
        body={mockBody}
        endpoint={endpoint}
        updatePoint={jest.fn()}
        updateState={jest.fn()}
        responseOption={'normal'}
        locations={mockLocations}
        selectedService={0}
        serviceOptions={serviceOptions}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
