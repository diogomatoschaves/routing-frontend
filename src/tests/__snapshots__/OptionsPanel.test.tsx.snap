// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render OptionsPanel component 1`] = `
<div
  className="sc-bdVaJa eBppGN"
>
  <div
    className="sc-bdVaJa fdCXKO"
    direction="row"
    height="70px"
  >
    <div
      className="sc-bdVaJa jJZREO"
      width="35%"
    >
      <div
        aria-expanded={false}
        className="ui fluid selection dropdown sc-EHOje jekteJ"
        id="selectedService"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Route
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Route
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Match
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa bJJihJ"
      width="35%"
    >
      <div
        className="sc-bdVaJa gxFsig"
        direction="row"
        height="45px"
      >
        <div
          className="sc-bdVaJa clGjkA"
          onClick={[Function]}
          width="10%"
        >
          <i
            aria-hidden="true"
            className="grey angle left large icon sc-bZQynM IAXmQ"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-bdVaJa gNhOwl"
          width="60%"
        >
          <div
            className="ui large label sc-gzVnrw fHkTB"
            onClick={[Function]}
          >
            No Traffic
          </div>
        </div>
        <div
          className="sc-bdVaJa clGjkA"
          onClick={[Function]}
          width="10%"
        >
          <i
            aria-hidden="true"
            className="grey angle right large icon sc-bZQynM IAXmQ"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="ui header sc-jTzLTM itsxkf"
      overridecolor="#00677f"
    >
      Request Endpoint
    </div>
    <div
      className="sc-bdVaJa fuiAsE"
      direction="row"
    >
      <div
        aria-expanded={false}
        className="ui fluid search selection dropdown sc-EHOje jekteJ"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <input
          aria-autocomplete="list"
          autoComplete="off"
          className="search"
          onChange={[Function]}
          style={
            Object {
              "width": undefined,
            }
          }
          tabIndex={0}
          type="text"
          value=""
        />
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          https://routing.develop.otonomousmobility.com/\${PROFILE}
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
          role="listbox"
        >
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.develop.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.staging.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.testing.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              http://localhost:5000
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
    >
      <div
        className="sc-bdVaJa kRVJAa"
        direction="row"
      >
        <div
          className="ui header sc-jTzLTM itsxkf"
          overridecolor="#00677f"
        >
          Request Body
        </div>
        <button
          className="ui button body-textarea sc-fjdhpX kqDxje"
          id="bodyEdit"
          marginleft="10px"
          onClick={[Function]}
        >
          Edit
        </button>
      </div>
    </div>
    <form
      className="ui form sc-gqjmRU dZmkkW"
      color="rgb(100, 100, 100)"
      focuscolor="#5097ab"
      onSubmit={[Function]}
    >
      <div
        className="sc-dnqmqq fNLREc"
      >
        <div
          className="sc-htoDjs epHUHX"
          onClick={[Function]}
        >
          
          <i
            aria-hidden="true"
            className="caret up icon"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-dnqmqq fNLREc"
        >
          <div
            className="sc-htoDjs epHUHX"
            onClick={[Function]}
          >
            Locations
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="sc-dnqmqq dROvRF"
        >
          <span
            className="sc-iwsKbI daDmXX"
          >
            Report Geometry
            : 
          </span>
          <span
            className="sc-gZMcBi hbrQes"
          >
            true
          </span>
        </div>
      </div>
    </form>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
    >
      <div
        className="ui header sc-jTzLTM itsxkf"
        overridecolor="#00677f"
      >
        Response
      </div>
      <button
        className="ui button response-textarea sc-fjdhpX kqDxje"
        id="responseEdit"
        marginleft="10px"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
    <form
      className="ui form sc-gqjmRU dZmkkW"
      color="rgb(100, 100, 100)"
      focuscolor="#5097ab"
      onSubmit={[Function]}
    >
      <div
        className="sc-dnqmqq fNLREc"
      >
        <div
          className="sc-htoDjs epHUHX"
          onClick={[Function]}
        >
          
          <i
            aria-hidden="true"
            className="caret up icon"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-dnqmqq fNLREc"
        >
          <span
            className="sc-iwsKbI daDmXX"
          >
            Code
            : 
          </span>
          <span
            className="sc-gZMcBi hbrQes"
          >
            Ok
          </span>
        </div>
        <div
          className="sc-dnqmqq fNLREc"
        >
          <div
            className="sc-htoDjs epHUHX"
            onClick={[Function]}
          >
            Routes
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="sc-dnqmqq dROvRF"
        >
          <div
            className="sc-htoDjs epHUHX"
            onClick={[Function]}
          >
            Locations
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
